graphql/resolvers.go

```diff
@@ -1850,10 +1851,48 @@ func (r *queryResolver) TaskTestSample(ctx context.Context, tasks []string, test
 			}
 			testResultsToReturn = append(testResultsToReturn, tr)
 		}
-		return testResultsToReturn, nil
+	} else {
+		regexFilter := ""
+		for i, f := range testFilters {
+			regexFilter += f.TestName
+			// If there is another test filter add a pipe to separate them
+			if i < len(testFilters)-1 {
+				regexFilter += "|"
+			}
+		}
+		for _, t := range dbTasks {
+			filteredTestResults, err := r.sc.FindTestsByTaskId(data.FindTestsByTaskIdOpts{
+				TaskID:    t.Id,
+				Execution: t.Execution,
+				TestName:  regexFilter,
+				Statuses:  []string{evergreen.TestFailedStatus},
```

Comment:
If we only ever filter on failed tests, can we just pass in a `[]string` of test name filters instead of `[]TestFilter`?

Reply:
This is meant to eventually support filtering on any status for different filters. So i opted to leave TestFilter as a struct. Since we will eventually need to add it. Leaving this as a struct will allow us to be able to make a non breaking change to the api when we do update this.